onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp

```diff
@@ -592,12 +593,129 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
         std::vector<DML_TENSOR_DESC> inputDescs = GetDmlInputDescs();
         std::vector<DML_TENSOR_DESC> outputDescs = GetDmlOutputDescs();
 
-        TOperatorDesc opDesc = {};
-        opDesc.InputTensor = &inputDescs[0];
-        opDesc.ScaleTensor = &inputDescs[1];
-        opDesc.ZeroPointTensor = hasZeroPointTensor ? &inputDescs[2] : nullptr;
-        opDesc.OutputTensor = &outputDescs[0];
-        SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+        if constexpr (std::is_same_v<TOperatorDesc, DML_QUANTIZE_OPERATOR_DESC> || std::is_same_v<TOperatorDesc, DML_DEQUANTIZE_OPERATOR_DESC>)
+        {
+            std::vector<DML_TENSOR_DESC> quantizationTensors;
+            quantizationTensors.push_back(inputDescs[1]);
+
+            const bool isSignedQuantization = inputDataType == DML_TENSOR_DATA_TYPE_INT4 ||
+                outputDataType == DML_TENSOR_DATA_TYPE_INT4 ||
+                inputDataType == DML_TENSOR_DATA_TYPE_INT8 ||
+                outputDataType == DML_TENSOR_DATA_TYPE_INT8;
+
+            if (hasZeroPointTensor || isSignedQuantization)
+            {
+                if (hasZeroPointTensor)
+                {
+                    quantizationTensors.push_back(inputDescs[2]);
+                }
+
+                TOperatorDesc opDesc = {};
+                opDesc.InputTensor = &inputDescs[0];
+                opDesc.QuantizationType = hasZeroPointTensor ? DML_QUANTIZATION_TYPE_SCALE_ZERO_POINT : DML_QUANTIZATION_TYPE_SCALE;
+                opDesc.QuantizationTensorCount = static_cast<uint32_t>(quantizationTensors.size());
+                opDesc.QuantizationTensors = quantizationTensors.data();
+                opDesc.OutputTensor = &outputDescs[0];
+                SetDmlOperatorDesc({ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &opDesc}, kernelInfo);
+            }
+            else
+            {
+                // For unsigned quantization, DML uses the midpoint of the datatype when zero point isn't provided (e.g. 8 for uint4 and 128 for uint8)
+                // since this is the most sane default in theory that represents the midpoint of the quantization. This is also the default used by various
+                // other frameworks and quantization tools. But because ONNX uses a default zero point of 0 no matter if it's signed or unsigned, we need
+                // to generate a constant zero point tensor with a value of 0 to override dml's default value when it isn't provided.
+                auto zeroPointDataType = std::is_same_v<TOperatorDesc, DML_QUANTIZE_OPERATOR_DESC> ? outputDataType : inputDataType;
+
+                // DML doesn't support int4 FILL_VALUE_CONSTANT yet, so simply create an int8 scalar and reinterpret it to an int4 tensor
```

Comment:
> DML doesn't support int4 FILL_VALUE_CONSTANT yet

(resolveme) That's going to be interesting, especially with existing output content and strides... (but yes, I think int4 should be supported eventually for the most primitive operations like fill and identity, even if via a cast and bitwise mask/or in the worst case).